core/src/main/java/com/spotify/metrics/net/NetstatMetricSet.java

```diff
@@ -0,0 +1,130 @@
+package com.spotify.metrics.net;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.codahale.metrics.Metric;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.spotify.metrics.core.DerivedLongGauge;
+import com.spotify.metrics.core.MetricId;
+import com.spotify.metrics.core.SemanticMetricSet;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class NetstatMetricSet implements SemanticMetricSet {
+
+    private static final String SPACES = "\\s+";
+    private static final String PROC_SNMP_PATH = "/proc/net/snmp";
+    private static final String PROC_NETSTAT_PATH = "/proc/net/netstat";
+
+    private final Supplier<Data> dataSupplier;
+
+    public NetstatMetricSet() {
+        this(Suppliers.memoizeWithExpiration(NetstatMetricSet::readData, 10, TimeUnit.SECONDS));
+    }
+
+    @VisibleForTesting
+    NetstatMetricSet(final Supplier<Data> dataSupplier) {
+        this.dataSupplier = checkNotNull(dataSupplier, "dataSupplier");
+    }
+
+    @Override
+    public Map<MetricId, Metric> getMetrics() {
+        final Map<MetricId, Metric> gauges = new HashMap<>();
+        gauges.put(
+            MetricId.build().tagged("what", "segments-retransmitted", "unit", "tcp-segments/s"),
```

Comment:
nitty-pick: as a lot of these are repeated string, would it make sense to create constants for them? They seem ok anyway, so feel free to ignore

Reply:
Personally for metrics I think it's easier to read spelled out like this.